sources/osgShader/Compiler.js

```diff
@@ -1012,22 +1000,9 @@ define( [
                     continue;
                 if ( !hasShadows ) {
                     hasShadows = true;
-                    this._vertexShader.push( 'vec4 worldPosition = ModelWorldMatrix * vec4(Vertex,1.0);' );
+                    this._vertexShader.push( 'WorldPosition = ModelWorldMatrix * vec4(Vertex,1.0);' );
```

Comment:
Would it be simpler to create a function hasShadowTexture that does the loop ... that would do something like that:
if (this.hasShadowTexture)
     this._vertexShader.push( 'WorldPosition = ModelWorldMatrix * vec4(Vertex,1.0);' );

it seems easier to read and maintain.